### PR TITLE
[FrameworkBundle] Run composer with --ignore-platform-req=php+

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
As described in https://github.com/composer/composer/issues/11068, we cannot use this flag on deps=low jobs.

**Comment by stof** [comment]
Shouldn't we use it only when testing unreleased PHP versions ?

**Comment by nicolas-grekas** [comment]
We strive to have deps (ours at least) use `>=` for the php constraint.

We should be using `--ignore-platform-req=php+` on all jobs for the same reason why we run with `--minimum-stability=dev`: to spot as early as possible when one of our deps needs help supporting an upcoming version of PHP.

**Comment by stof** [comment]
Using `--ignore-platform-req=php+` on stable versions of PHP will not help that (what helps that is always having a job running against PHP nightly, which would indeed need to use that option).
On the other hand, using `--ignore-platform-req=php+` on stable versions would *hide* the fact that some deps are marked as incompatible with existing stable PHP version, which would then make them unusable in projects unless they use `--ignore-platform-req=php+` themselves.

**Comment by nicolas-grekas** [comment]
ACK, updated. I saw that config on doctrine/migrations, they might want to adjust /cc @derrabus

File: src/Symfony/Component/Lock/composer.json [hunk 1]
```diff
@@ -21,11 +21,11 @@
         "symfony/polyfill-php80": "^1.16"
     },
     "require-dev": {
-        "doctrine/dbal": "^2.6|^3.0",
+        "doctrine/dbal": "^2.7|^3.0",
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Lock/composer.json:24]: just syncing with the constraint on the root composer.json

**Review by stof — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Comment by derrabus** [comment]
I think, in doctrine/migrations this setting is just fine. I'd like to keep the build simple there.

**Comment by nicolas-grekas** [comment]
> I think, in doctrine/migrations this setting is just fine. I'd like to keep the build simple there.

Fine to me; just know that the deps=low job installs never tested combinations.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**